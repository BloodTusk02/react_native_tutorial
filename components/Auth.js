import React from "react";
import { View, Text, StyleSheet, TouchableOpacity } from "react-native";

const Auth = ({ navigation }) => (
  <View style={{ flex: 1 }}>
    <View style={{ ...styles.container, backgroundColor: "white" }}>
      <Text style={{ fontSize: 25 }}>Войдите в приложение</Text>
      <View style={{ marginVertical: 10 }}>
        <TouchableOpacity
          style={{ ...styles.button, marginVertical: 50, marginBottom: 10 }}
          onPress={() => {
            navigation.navigate("Login");
          }}
        >
          <Text style={styles.btnText}>Log In</Text>
        </TouchableOpacity>
      </View>
      <TouchableOpacity
        style={styles.button}
        onPress={() => {
          navigation.navigate("Registration");
        }}
      >
        <Text style={styles.btnText}>Register</Text>
      </TouchableOpacity>
      <TouchableOpacity
        style={{...styles.button, marginTop: 20}}
        onPress={() => {
          navigation.navigate("CreateForm");
        }}
      >
        <Text style={styles.btnText}>Create Form</Text>
      </TouchableOpacity>
      <TouchableOpacity
        style={{...styles.button, marginTop: 20}}
        onPress={() => {
          navigation.navigate("MainScreen");
        }}
      >
        <Text style={styles.btnText}>MainScreen</Text>
      </TouchableOpacity>
    </View>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
  button: {
    backgroundColor: "#F57600",
    height: 50,
    width: 290,
    alignItems: "center",
    justifyContent: "center",
    borderWidth: 2,
    borderRadius: 10,
    borderColor: "#F57600",
  },
  btnText: {
    color: "white",
    fontSize: 25,
  },
});

export default Auth;
