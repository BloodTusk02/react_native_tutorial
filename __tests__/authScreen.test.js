import React from 'react';
import renderer from 'react-test-renderer';
import Auth from '../components/Auth';

test('renders correctly', () => {
    const tree = renderer.create(<Auth />).toJSON();
    expect(tree).toMatchSnapshot();
  });