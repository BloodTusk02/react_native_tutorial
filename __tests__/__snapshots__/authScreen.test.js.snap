// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 25,
        }
      }
    >
      Войдите в приложение
    </Text>
    <View
      style={
        Object {
          "marginVertical": 10,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F57600",
            "borderColor": "#F57600",
            "borderRadius": 10,
            "borderWidth": 2,
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginVertical": 50,
            "opacity": 1,
            "width": 290,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 25,
            }
          }
        >
          Log In
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F57600",
          "borderColor": "#F57600",
          "borderRadius": 10,
          "borderWidth": 2,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": 290,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 25,
          }
        }
      >
        Register
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F57600",
          "borderColor": "#F57600",
          "borderRadius": 10,
          "borderWidth": 2,
          "height": 50,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "width": 290,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 25,
          }
        }
      >
        Create Form
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F57600",
          "borderColor": "#F57600",
          "borderRadius": 10,
          "borderWidth": 2,
          "height": 50,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "width": 290,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 25,
          }
        }
      >
        MainScreen
      </Text>
    </View>
  </View>
</View>
`;
